repository/src/main/java/org/apache/atlas/repository/store/aliasstore/ESAliasStore.java

```diff
@@ -207,10 +209,11 @@ private void personaPolicyToESDslClauses(List<AtlasEntity> policies,
                     terms.add(connectionQName);
 
                 } else if (getPolicyActions(policy).contains(ACCESS_READ_PERSONA_GLOSSARY)) {
-
-                    for (String glossaryQName : assets) {
-                        terms.add(glossaryQName);
-                        allowClauseList.add(mapOf("wildcard", mapOf(QUALIFIED_NAME, "*@" + glossaryQName)));
+                    List<String> glossaryQualifiedNames = assets.stream()
+                            .peek(terms::add)
+                            .collect(Collectors.toList());
+                    if (!glossaryQualifiedNames.isEmpty()) {
+                        allowClauseList.add(mapOf("terms", mapOf(GLOSSARY_QUALIFIED_NAME_ATTRIBUTE, glossaryQualifiedNames)));
```

Comment:
Simplify this

Reply:
```
if (CollectionUtils.isNotEmpty(assets)) {
    terms.addAll(assets);
    allowClauseList.add(mapOf("terms", mapOf(GLOSSARY_PROPERTY_KEY, assets)));
}

Reply:
Forgot that assets is list of qualifiedName :)